cli/src/commands/bookmark/list.rs

```diff
@@ -277,3 +307,492 @@ struct RefListItem {
+        bookmark_items.sort_by_key(|item| {
+            (
+                item.primary.name().to_owned(),
+                item.primary.remote_name().map(|name| name.to_owned()),
+            )
+        });
+
+        sort_and_snapshot(&mut bookmark_items, sort_keys, &commits)
+    }
+
+    // Helper function to sort refs and prepare snapshot with relevant information.
+    fn sort_and_snapshot(
+        items: &mut [RefListItem],
+        sort_keys: &[SortKey],
+        commits: &HashMap<CommitId, Arc<backend::Commit>>,
+    ) -> String {
+        sort(items, sort_keys, commits);
+
+        let to_commit = |item: &RefListItem| {
+            let id = item.primary.target().added_ids().next()?;
+            commits.get(id)
+        };
+        items
+            .iter()
+            .map(|item| {
+                sort_keys
+                    .iter()
+                    .map(|key| match key {
+                        SortKey::Name | SortKey::NameDesc => {
+                            [Some(item.primary.name()), item.primary.remote_name()]
+                                .iter()
+                                .flatten()
+                                .join("@")
+                        }
+                        SortKey::AuthorName | SortKey::AuthorNameDesc => to_commit(item)
+                            .map(|commit| format!("author: {}", commit.author.name))
+                            .unwrap_or_default(),
+                        SortKey::AuthorEmail | SortKey::AuthorEmailDesc => to_commit(item)
+                            .map(|commit| format!("author.email: {}", commit.author.email))
+                            .unwrap_or_default(),
+                        SortKey::AuthorDate | SortKey::AuthorDateDesc => to_commit(item)
+                            .map(|commit| {
+                                format!("author.timestamp: {}", commit.author.timestamp.timestamp.0)
+                            })
+                            .unwrap_or_default(),
+                        SortKey::CommitterName | SortKey::CommitterNameDesc => to_commit(item)
+                            .map(|commit| format!("committer: {}", commit.committer.name))
+                            .unwrap_or_default(),
+                        SortKey::CommitterEmail | SortKey::CommitterEmailDesc => to_commit(item)
+                            .map(|commit| format!("committer.email: {}", commit.committer.email))
+                            .unwrap_or_default(),
+                        SortKey::CommitterDate | SortKey::CommitterDateDesc => to_commit(item)
+                            .map(|commit| {
+                                format!(
+                                    "committer.timestamp: {}",
+                                    commit.committer.timestamp.timestamp.0
+                                )
+                            })
+                            .unwrap_or_default(),
+                    })
+                    .join(", ")
```

Comment:
nit: I think it's simpler to just print all columns. The output is pretty readable if columns are aligned.